src/credentials/CHIPOpCred.h

```diff
@@ -0,0 +1,184 @@
+     **/
+    const ChipCertificateSet * GetOpCredSet() const { return mOpCreds; }
+
+    /**
+     * @return A pointer to the last certificate-set in the set. Returns NULL if certificate set is empty.
+     **/
+    const ChipCertificateSet * GetLastCert() const { return (mOpCredCount > 0) ? &mOpCreds[mOpCredCount - 1] : nullptr; }
+
+    /**
+     * @return Number of certificate-sets loaded into the set.
+     **/
+    uint8_t GetCertCount() const { return mOpCredCount; }
+
+    /**
+     * @brief Retrieve Trusted Root ID / Authority Key ID of a Certificate Set.
+     *
+     * @param certSetIndex  Index of the Certificate-set to be used to search for a Trusted Root ID.
+     *
+     * @return A pointer to the Trusted Root ID on success. Otherwise, nullptr if no Trust Anchor is found.
+     **/
+    const CertificateKeyId * GetTrustedRootId(uint16_t certSetIndex) const;
+
+    /**
+     * @brief Check whether certificate set is in the operational credential set.
+     *
+     * @param cert  Pointer to the ChipCertificateSet structures.
+     *
+     * @return True if certificate is in the set, false otherwise.
+     **/
+    bool IsCertSetInTheOpCredSet(const ChipCertificateSet * cert) const;
+
+    /**
+     * @brief Check whether Trusted Root ID is in the operational credential set.
+     *
+     * @param trustedRoot  Reference to the Trusted Root ID.
+     *
+     * @return True if Trusted Root ID is in the set, false otherwise.
+     **/
+    bool IsTrustedRootIn(const CertificateKeyId & trustedRoot) const;
+
+    /**
+     * @brief Validate CHIP certificate.
+     *
+     * @param trustedRootId    Reference to the Trusted Root ID for the Certificate Set to be used
+     *                          for validation
+     * @param cert              Pointer to the CHIP certificiate to be validated. The certificate is
+     *                          required to be in this set, otherwise this function returns error.
+     * @param context           Certificate validation context.
+     *
+     * @return Returns a CHIP_ERROR on validation or other error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR ValidateCert(const CertificateKeyId & trustedRootId, const ChipCertificateData * cert, ValidationContext & context);
+
+    /**
+     * @brief Find and validate CHIP certificate.
+     *
+     * @param trustedRootId    Reference to the Trusted Root ID for the Certificate Set to be used
+     *                          for validation
+     * @param subjectDN         Subject distinguished name to use as certificate search parameter.
+     * @param subjectKeyId      Subject key identifier to use as certificate search parameter.
+     * @param context           Certificate validation context.
+     * @param cert              A pointer to the valid CHIP certificate that matches search criteria.
+     *
+     * @return Returns a CHIP_ERROR on validation or other error, CHIP_NO_ERROR otherwise
```

Comment:
how about 'Not found'? I imagine this would rather be 'returns an error if no valid certificate could be found'.

But this is obvious: a method returning CHIP_ERROR is expected to return error on failure and NO_ERROR on success. Can we say something non-obvious in the comments? Like is cert expected to be nullptr on error or is the return value undefined if error is returned?